src/Agent.Listener/Agent.cs

```diff
@@ -35,10 +34,16 @@ public override void Initialize(IHostContext hostContext)
 
         public async Task<int> ExecuteCommand(CommandSettings command)
         {
+            AssemblyLoadContext loadContext = AssemblyLoadContext.GetLoadContext(typeof(Agent).GetTypeInfo().Assembly);
```

Comment:
This busts the unit testing pattern. Can you move the Unloading event to IHostContext? It can just be a simple event handler (`AssemblyLoadContext obj` is not required since it's unused anyway).